changelogs/fragments/drop-host-api-fallback.yml

```diff
@@ -0,0 +1,2 @@
+bugfixes:
```

Comment:
I wasn't sure about this. Conceptually it's a bugfix (you really do not want wrong data when running stuff), but it's also technically a breaking change :see_no_evil: 

Realistically, this is only a breaking change for users who
- do not have `foreman_ansible` installed
- have `use_reports_api: true`
- have ignored the "Failed to use Reports API, falling back to Hosts API" warning ever since
:woman_shrugging:

Reply:
And additionally if that thing ever triggered, there inventory ran into the wrong data (format) and presumably failed anyway down the line? IMHO It definitely is a bugfix. It breaks bug to bug compatibility, right?